servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/DefaultGrpcClientCallFactory.java

```diff
@@ -155,7 +156,10 @@
                 newBlockingStreamingCall(serializationProvider, requestClass, responseClass);
         return (metadata, request) -> {
             try (BlockingIterator<Resp> iterator = streamingClientCall.request(metadata, request).iterator()) {
-                final Resp firstItem = iterator.next();
+                final Resp firstItem;
+                if (!iterator.hasNext() || (firstItem = iterator.next()) == null) {
```

Comment:
`null` value here is essentially OK from the context of this call and we generally pass through `null`.

Reply:
Can you clarify in which situations we expect to return a `null` value for a scalar blocking RPC call?

Reply:
The scalar/streaming blocking/async variant is immaterial here and no we do not expect a `null` value but neither do we use the `Resp` here in a way that it can not be `null`. 

Hypothesizing situations, `streamingClientCall` that is giving the `BlockingIterator` here is created using a `StreamingHttpClient` that is provided by the user of `GrpcClientCallFactory` API. It is nowhere we check that the payload of the response returned by the client can not be `null` unless we actually use it, eg: while deserializing. `Deserializer` here also is provided by the user. So really a user can pass `null` from everywhere and get back a `null` response and we really do not care.

So the point here isn't whether it is fair to make that assumption, it is really about whether there is any positive of making that assumption. Same argument as passing `null` through async sources, we do not care about it being `null` unless we use it.